### DISCUSSION TITLE
binary_sensor.octopus_energy_target_my_charge_slots

### DISCUSSION THREAD

**Discussion by Dave-Parry** [BottlecapDave/HomeAssistant-OctopusEnergy#704] [General]
I have just got a unavailable icon when I look in entities for binary_sensor.octopus_energy_target_my_charge_slots.
The Time slot display seems to be working ok, but was wondering why its just appeared.

**Comment by BottlecapDave** [comment]
Hello. Sorry for the late response. Did you have any errors in your logs?

**Comment by Dave-Parry** [comment]
I found this log. Does that help?
Regards
David Parry
***@***.***

￼

**Reply by BottlecapDave** [reply]
Hello. I don't see any logs attached to this? Github has issues with attachments to email replies

**Comment by Dave-Parry** [comment]
Found out I can copy the log so here it is.

Logger: homeassistant.helpers.template
Source: helpers/template.py:2318
First occurred: 21 January 2024 at 16:57:57 (6753 occurrences)
Last logged: 09:42:58

Template variable error: 'dict object' has no attribute 'valid_from' when rendering '{% if state_attr('binary_sensor.octopus_energy_target_my_charge_slots', 'target_times') | length > 0 %} {% for target_time in state_attr('binary_sensor.octopus_energy_target_my_charge_slots', 'target_times') -%} ⏰ {{ (target_time.valid_from | as_local).strftime('%H:%M') }} to {{ (target_time.valid_to | as_local).strftime('%H:%M') }} {{ "\n" }} {%- endfor %} {%- else -%} ⏰ None {%- endif %}'

Regards
David Parry
***@***.***

**Reply by BottlecapDave** [reply]
Assuming you're on the latest version of the integratio, that template structure has not been valid since v9. `valid_from` is now `start` and `valid_to` is now `end`.

**Comment by Dave-Parry** [comment]
Ok, Thanks Dave,
Do I just leave it or is there a way to delete it?

Regards
David Parry
***@***.***

**Comment by Dave-Parry** [comment]
forget that Dave, just deleted it.

Regards
David Parry
***@***.***
